lib/components/Modal/Modal.test.tsx

```diff
@@ -1,22 +1,29 @@
-import { render, screen } from '@testing-library/react';
-import { expect, test, vi } from 'vitest';
+import { act, render, screen } from '@testing-library/react';
+import { describe, expect, it, vi } from 'vitest';
 import { Modal } from './Modal';
 
-test('Modal renders correctly', () => {
-  const content = <div>Modal Content</div>;
-  const footer = <div>Modal Footer</div>;
+describe('Modal', () => {
+  it('renders correctly', async () => {
+    const content = <div>Modal Content</div>;
+    const footer = <div>Modal Footer</div>;
 
-  render(<Modal open={true} onClose={vi.fn()} content={content} footer={footer} />);
+    // While the Modal component is not async, doing "await act" will suppress the
+    // "An update inside a test was not wrapped in act(...)." warning
+    // eslint-disable-next-line @typescript-eslint/require-await
+    await act(async () => {
+      render(<Modal open={true} onClose={vi.fn()} content={content} footer={footer} />);
+    });
 
-  // Assert that the modal is rendered
-  const modalElement = screen.getByRole('dialog');
-  expect(modalElement).toBeInTheDocument();
+    // Assert that the modal is rendered
+    const modalElement = screen.getByRole('dialog');
+    expect(modalElement).toBeInTheDocument();
```

Comment:
I tested and it seems that only change that worked for me was to use `await waitFor`. So no need for eslint-disable or await act sync.
```tsx
//...
import { render, screen, waitFor } from '@testing-library/react';
//...
    await waitFor(() => expect(modalElement).toBeInTheDocument());
```

In this case needed for only the first.